### PR TITLE
TASK: Use NodeAddressFactory for NodeAddress creation

### PR DESCRIPTION
**Pull request by dlubitz**

### CODE REVIEW

File: Neos.Neos/Classes/FrontendRouting/NodeAddressFactory.php [hunk 1]
```diff
@@ -36,23 +36,35 @@ public static function create(ContentRepository $contentRepository): self
         return new self($contentRepository);
     }
 
-    public function createFromNode(Node $node): NodeAddress
-    {
+    public function createFromContentStreamIdAndDimensionSpacePointAndNodeAggregateId(
```
Comment (mhsdesign) [inline Neos.Neos/Classes/FrontendRouting/NodeAddressFactory.php:39]: 😂 good name ^^ but actually makes sense to have.

For the future (as the content repository id will be added also)

we should consider using only two arguments:

- NodeAggregateId
and
- SubgraphIdentity

```
public function createFromSubgraphIdentityAndNodeAggregateId
```

Comment (dlubitz) [reply Neos.Neos/Classes/FrontendRouting/NodeAddressFactory.php:39]: Huu? But we don't have the subgraph there.

Comment (mhsdesign) [reply Neos.Neos/Classes/FrontendRouting/NodeAddressFactory.php:39]: but we can create it? https://github.com/neos/neos-development-collection/blob/5f80a56210ba903e484e79a99d82b457a226198f/Neos.ContentRepository.Core/Classes/Projection/ContentGraph/ContentSubgraphIdentity.php#L40-L47

Comment (dlubitz) [reply Neos.Neos/Classes/FrontendRouting/NodeAddressFactory.php:39]: Hmm.... but we already have the contentrepository in the factory. So we would need to create a object with two more (and unneeded) objects.

Comment (dlubitz) [reply Neos.Neos/Classes/FrontendRouting/NodeAddressFactory.php:39]: And which contentRepository is used then? This from factory or this from SubgraphIdentity?

Comment (mhsdesign) [reply Neos.Neos/Classes/FrontendRouting/NodeAddressFactory.php:39]: ah i see, i thought we didnt had the information ... id say assert they are the same. (Same dilemma for the `forNode` method.

Comment (dlubitz) [reply Neos.Neos/Classes/FrontendRouting/NodeAddressFactory.php:39]: I'd keep it as it is. Doesn't make sence to wrap the parameter with two unsed ones to save one on the other side.

Comment (skurfuerst) [reply Neos.Neos/Classes/FrontendRouting/NodeAddressFactory.php:39]: lets keep as proposed here :+1:

**Review by skurfuerst — APPROVED** [review]
thanks!

**Review by bwaidelich — APPROVED** [review]
I like, +1 by reading <3

**Pull request merged**